cachi2/core/package_managers/cargo/main.py

```diff
@@ -127,20 +130,24 @@ def _resolve_cargo_package(
     """Resolve a single cargo package."""
     _verify_lockfile_is_present_or_fail(package_dir)
     vendor_dir = output_dir.join_within_root("deps/cargo")
-    cmd = ["cargo", "vendor", "--locked", str(vendor_dir)]
+    # --no-delete to keep everything already present. It does not matter for a fresh
+    # single package, but it does matter when there is pip interaction.
+    cmd = ["cargo", "vendor", "--locked", "--versioned-dirs", "--no-delete", str(vendor_dir)]
     log.info("Fetching cargo dependencies at %s", package_dir)
     with _hidden_cargo_config_file(package_dir):
         # stdout contains exact values to add to .cargo/config.toml for a build to become hermetic.
         config_template = run_cmd(cmd=cmd, params={"cwd": package_dir})
 
     packages = _extract_package_info(package_dir.path / "Cargo.lock")
     main_package = _resolve_main_package(package_dir)
-    vcs_url = get_repo_id(package_dir.root).as_vcs_url_qualifier()
-    is_a_dep = lambda p: p["name"] != main_package["name"]
-    deps_components = (
-        CargoPackage(**p, vcs_url=vcs_url).to_component() for p in packages if is_a_dep(p)
-    )
+    is_a_dep = lambda p: p["name"] != main_package["name"]  # a shorthand, thus # noqa: E731
+    deps_components = (CargoPackage(**p).to_component() for p in packages if is_a_dep(p))
```

Comment:
This looks like an incorrectly resolved merge conflict, will fix in the next revision.